IMX targets currently require using u-boot's private libgcc, or they
FTBFS with Debian's gcc toolchain. It's not ideal, and a thread was
started on the u-boot list discussing the issue:
http://lists.denx.de/pipermail/u-boot/2016-March/249124.html

Index: u-boot/configs/mx53loco_defconfig
===================================================================
--- u-boot.orig/configs/mx53loco_defconfig
+++ u-boot/configs/mx53loco_defconfig
@@ -14,3 +14,4 @@ CONFIG_CMD_EXT2=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/mx6cuboxi_defconfig
===================================================================
--- u-boot.orig/configs/mx6cuboxi_defconfig
+++ u-boot/configs/mx6cuboxi_defconfig
@@ -22,3 +22,4 @@ CONFIG_CMD_FS_GENERIC=y
 CONFIG_DM=y
 CONFIG_DM_THERMAL=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/nitrogen6q_defconfig
===================================================================
--- u-boot.orig/configs/nitrogen6q_defconfig
+++ u-boot/configs/nitrogen6q_defconfig
@@ -33,3 +33,4 @@ CONFIG_G_DNL_MANUFACTURER="Boundary"
 CONFIG_G_DNL_VENDOR_NUM=0x0525
 CONFIG_G_DNL_PRODUCT_NUM=0xa4a5
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/novena_defconfig
===================================================================
--- u-boot.orig/configs/novena_defconfig
+++ u-boot/configs/novena_defconfig
@@ -27,3 +27,4 @@ CONFIG_USB=y
 CONFIG_USB_GADGET=y
 CONFIG_CI_UDC=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/udoo_defconfig
===================================================================
--- u-boot.orig/configs/udoo_defconfig
+++ u-boot/configs/udoo_defconfig
@@ -20,3 +20,4 @@ CONFIG_CMD_FS_GENERIC=y
 CONFIG_DM=y
 CONFIG_DM_THERMAL=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/usbarmory_defconfig
===================================================================
--- u-boot.orig/configs/usbarmory_defconfig
+++ u-boot/configs/usbarmory_defconfig
@@ -17,3 +17,4 @@ CONFIG_CMD_EXT4=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
Index: u-boot/configs/wandboard_defconfig
===================================================================
--- u-boot.orig/configs/wandboard_defconfig
+++ u-boot/configs/wandboard_defconfig
@@ -23,3 +23,4 @@ CONFIG_CMD_FS_GENERIC=y
 CONFIG_DM=y
 CONFIG_DM_THERMAL=y
 CONFIG_OF_LIBFDT=y
+CONFIG_USE_PRIVATE_LIBGCC=y
